ui/src/App.tsx

```diff
@@ -98,32 +136,86 @@ const connectWallet = async () => {
   }
 };
 
-const makeOffer = () => {
-  const { wallet, contractInstance, brands, issuers } = useAppStore.getState();
-  const istBrand = brands?.find(([name]) => name === 'IST')?.at(1);
-  const istIssuer = issuers?.find(([name]) => name === 'IST')?.at(1);
-  const bldBrand = brands?.find(([name]) => name === 'BLD')?.at(1);
-  const bldIssuer = issuers?.find(([name]) => name === 'BLD')?.at(1);
+const ConnectWallet = () => (
+  <div className="card">
+    <button onClick={connectWallet}>Connect Wallet</button>
+  </div>
+);
+
+const useForm = create<FormState>(() => ({
+  recipient: recipientAddr,
+  give: undefined,
+  want: undefined,
+}));
+
+const setFirstOffer = ({ recipient = undefined }) => {
+  console.log('ADDRESS', recipient);
+  const { brands } = useAppStore.getState();
+
+  const istBrand = brands?.get('IST');
+  const bldBrand = brands?.get('BLD');
+
+  useForm.setState({
+    recipient,
+    give: { Price: { brand: istBrand, value: 15_000_000n }, Fee: { brand: istBrand, value: 1_000_000n } },
+    want: { Value: { brand: bldBrand, value: 1_000_000n } },
+  });
+};
 
-  const value = makeCopyBag([
-    ['FTX Arena', 2n],
-    ['Crypto.com Arena', 1n],
-  ]);
+const setMatchOffer = (invite: Invitation) => {
+  const { give, want } = invite.customDetails;
+  console.log('MATCH', give, want);
+  // what if there isn't a Fee?
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { Fee, ...withoutFee } = give;
+  useForm.setState({
+    recipient: undefined,
+    give: want,
+    want: withoutFee,
+  });
+};
+
+
+const makeSecondOffer = () => {
+  const { wallet, contractInstance, } = useAppStore.getState();
+  const { give, want } = useForm.getState();
+  wallet?.makeOffer(
+    {
+      source: 'purse',
+      instance: contractInstance,
+      description: 'matchOffer'
+    },
+    { give, want },
+    undefined,
+    (update: { status: string; data?: unknown }) => {
+      if (update.status === 'error') {
+        alert(`Match offer error: ${update.data}`);
+      }
+      if (update.status === 'accepted') {
+        alert('Match offer accepted');
+      }
+      if (update.status === 'refunded') {
+        alert('Match offer rejected');
+      }
+    }
+  );
+};
 
-  // const give = { Price: { brand: istBrand, value: 15_000_000n } };
-  const give = { Price: { brand: istBrand, value: 15_000_000n }, Fee: { brand: istBrand, value: 1_000_000n } };
-  const want = { Value: { brand: bldBrand, value: 1_000_000n } };
+const makeOffer = () => {
+  const { wallet, contractInstance, } = useAppStore.getState();
+  const { recipient, give, want } = useForm.getState();
+  console.log('ADDRESS', recipient);
 
   wallet?.makeOffer(
     {
       source: 'contract',
       instance: contractInstance,
       publicInvitationMaker: 'makeFirstInvitation',
       // HACK setup a trade
```

Comment:
This comment doesn't explain much, maybe give an idea of what would ideally be included in `invitationArgs`?